pkg/cluster/volumes.go

```diff
@@ -151,7 +151,7 @@ func (c *Cluster) populateVolumeMetaData() error {
 	volumeIds := []string{}
 	var volumeID string
 	for _, pv := range pvs {
-		volumeID, err = c.VolumeResizer.ExtractVolumeID(pv.Spec.AWSElasticBlockStore.VolumeID)
+		volumeID, err = c.VolumeResizer.GetProviderVolumeID(pv)
```

Comment:
calling GetProviderVolumeID is fine with we since it will ultimately also call ExtractVolumeID. However, for some unit test you have to add the mocked reponses of the resizer. Before [these lines](https://github.com/zalando/postgres-operator/blob/master/pkg/cluster/volumes_test.go#L216) you would need to `resizer.EXPECT().GetProviderVolumeID(gomock.Eq(...`

Reply:
I just pushed a fix to this. Let me know if it's ok. I'm pretty new to Go and I have been away from typed languages for a long time, so I could miss some tricks and practices.